launcher.c

```diff
@@ -168,26 +168,17 @@ char **parse_argv(char *cmdline, int *argc)
     } while (1);
 }
 
-void pass_control_to_child(DWORD control_type) {
-    /*
-     * distribute-issue207
-     * passes the control event to child process (Python)
-     */
-    if (!child_pid) {
-        return;
-    }
-    GenerateConsoleCtrlEvent(child_pid,0);
-}
-
 BOOL control_handler(DWORD control_type) {
     /* 
      * distribute-issue207
-     * control event handler callback function
+     * control event handler callback function and
+     * pass the control event to child process.
      */
-    switch (control_type) {
-        case CTRL_C_EVENT:
-            pass_control_to_child(0);
-            break;
+    if ((control_type == CTRL_C_EVENT) && child_pid){
+        if (!GenerateConsoleCtrlEvent(child_pid, 0)){
```

Comment:
```suggestion
        if (!GenerateConsoleCtrlEvent(CTRL_C_EVENT, 0)){
```

Reply:
https://docs.microsoft.com/en-us/windows/console/generateconsolectrlevent

> BOOL WINAPI GenerateConsoleCtrlEvent(
  _In_ DWORD dwCtrlEvent,
  _In_ DWORD dwProcessGroupId
);
 If dwProcessGroupId is nonzero, this function will succeed, but the CTRL+C signal will not be received by processes within the specified process group.

Reply:
Well, this got me to look at what is done in distlib https://bitbucket.org/pypa/distlib/src/3a30409dd232f52390f55bb80734e80be1d392af/PC/launcher.c#lines-430 (it was me who copied from that)
as well as in `py.exe` launcher https://github.com/python/cpython/blob/e5c7ee1156986c2046799ce523f0df54af51b7a3/PC/launcher.c#L740-L744
and from `py.exe` launcher it feels that it is better to ignore those control events and wait for `cmd` to kill or send events to python first and exit with the same return code as done in `py.exe` also, this should be done in distlib also.

Reply:
I have fixed it.